qiskit/circuit/library/standard_gates/ryy.py

```diff
@@ -75,38 +74,27 @@ def __init__(self, theta):
 
     def _define(self):
         """Calculate a subcircuit that implements this unitary."""
-        from .x import CXGate
-        from .rx import RXGate
-        from .rz import RZGate
-
-        definition = []
-        q = QuantumRegister(2, 'q')
-        theta = self.params[0]
-        rule = [
-            (RXGate(np.pi / 2), [q[0]], []),
-            (RXGate(np.pi / 2), [q[1]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (RZGate(theta), [q[1]], []),
-            (CXGate(), [q[0], q[1]], []),
-            (RXGate(-np.pi / 2), [q[0]], []),
-            (RXGate(-np.pi / 2), [q[1]], []),
-        ]
-        for inst in rule:
-            definition.append(inst)
-        self.definition = definition
+        circ = self.decompositions[0]
+        phase = circ.phase / len(circ.qregs[0])
+        if circ.phase:
+            circ.u3(np.pi, phase, phase - np.pi, circ.qregs[0])
+            circ.x(circ.qregs[0])
```

Comment:
adding two gates just to fix a phase seems very inefficient. phase should just be a number.
probably the reason for this is that `definition` is not an actual circuit to support phase. if so, then I think we need to rethink how definition is done. I think @kdk wanted to remove them altogether.

Reply:
This was meant as a temporary way of allowing conversion of a circuit with a global phase to a gate. I agree there should be something more efficient. I think perhaps once definition becomes a circuit this would not be necessary but I wanted to leave that for a separate PR  since it might be a big one.

Reply:
Agree there's a need for a follow up PR to either a) convert `definition` to a circuit or b) (preferred) update `ControlledGate` to read from `decompositions`. For this PR, the longer but phase correct definitions are the only way to get correct phases on `ControlledGate`s. (And the added gates for phase correction should only impact users not running `Optimize1qGates` or using `BasisTranslator`.)

Reply:
now that #4622 has landed we can add a phase whenever a decomposition/definition is not exactly equal to the gate matrix.